pkg/uploadserver/uploadserver_test.go

```diff
@@ -204,209 +200,175 @@ func replaceAsyncProcessorFunc(replacement func(io.ReadCloser, string, string, s
-	req, err := http.NewRequest("GET", healthzPath, nil)
-	if err != nil {
-		t.Fatal(err)
-	}
+	It("healthz", func() {
+		req, err := http.NewRequest("GET", healthzPath, nil)
+		Expect(err).ToNot(HaveOccurred())
 
-	rr := httptest.NewRecorder()
+		rr := httptest.NewRecorder()
 
-	app := uploadServerApp{}
-	server, _ := app.createHealthzServer()
-	server.Handler.ServeHTTP(rr, req)
+		app := uploadServerApp{}
+		server, _ := app.createHealthzServer()
+		server.Handler.ServeHTTP(rr, req)
 
-	if status := rr.Code; status != http.StatusOK {
-		t.Errorf("handler returned wrong status code: got %v want %v",
-			status, http.StatusOK)
-	}
-}
+		status := rr.Code
+		Expect(status).To(Equal(http.StatusOK))
 
-func TestInProcessUnavailable(t *testing.T) {
-	withProcessorSuccess(func() {
-		req := newRequest(t)
+	})
 
-		rr := httptest.NewRecorder()
+	It("Process unavailable", func() {
+		withProcessorSuccess(func() {
+			req := newRequest()
+
+			rr := httptest.NewRecorder()
 
-		server := newServer()
-		server.uploading = true
-		server.ServeHTTP(rr, req)
+			server := newServer()
+			server.uploading = true
+			server.ServeHTTP(rr, req)
+
+			status := rr.Code
+			Expect(status).To(Equal(http.StatusServiceUnavailable))
+		})
 
-		if status := rr.Code; status != http.StatusServiceUnavailable {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusServiceUnavailable)
-		}
 	})
-}
 
-func TestInProcessUnavailableAsync(t *testing.T) {
-	withProcessorSuccess(func() {
-		req := newAsyncRequest(t)
+	It("Process unavailable, async", func() {
+		withProcessorSuccess(func() {
+			req := newAsyncRequest()
+
+			rr := httptest.NewRecorder()
 
-		rr := httptest.NewRecorder()
+			server := newServer()
+			server.uploading = true
+			server.ServeHTTP(rr, req)
 
-		server := newServer()
-		server.uploading = true
-		server.ServeHTTP(rr, req)
+			status := rr.Code
+			Expect(status).To(Equal(http.StatusServiceUnavailable))
+		})
 
-		if status := rr.Code; status != http.StatusServiceUnavailable {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusServiceUnavailable)
-		}
 	})
-}
-func TestCompletedConflict(t *testing.T) {
-	withProcessorSuccess(func() {
-		req := newRequest(t)
 
-		rr := httptest.NewRecorder()
+	It("Completed conflict", func() {
+		withProcessorSuccess(func() {
+			req := newRequest()
+
+			rr := httptest.NewRecorder()
 
-		server := newServer()
-		server.done = true
-		server.ServeHTTP(rr, req)
+			server := newServer()
+			server.done = true
+			server.ServeHTTP(rr, req)
 
-		if status := rr.Code; status != http.StatusConflict {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusConflict)
-		}
+			status := rr.Code
+			Expect(status).To(Equal(http.StatusConflict))
+		})
 	})
-}
 
-func TestCompletedConflictAsync(t *testing.T) {
-	withProcessorSuccess(func() {
-		req := newAsyncRequest(t)
+	It("Completion conflict, async", func() {
```

Comment:
Only difference between this one and the one above is the request type, can use a table with the request as a parameter to reduce code. Maybe combine the above 4 into a single table if you pass uploading and done values as a parameter to the table function.